internal/cli/atlas/backup/schedule/update.go

```diff
@@ -43,35 +56,168 @@ func (opts *UpdateOpts) initStore(ctx context.Context) func() error {
+	out.UseOrgAndGroupNamesInExportPrefix = returnValueForSetting(opts.useOrgAndGroupNamesInExportPrefix, opts.noUseOrgAndGroupNamesInExportPrefix)
+
+	return out, nil
+}
+
+func (opts *UpdateOpts) verifyExportBucketID(out *atlas.CloudProviderSnapshotBackupPolicy) {
+	if opts.exportBucketID != "" {
+		checkForExport(out)
+		out.Export.ExportBucketID = opts.exportBucketID
+	}
+}
+
+func (opts *UpdateOpts) verifyExportFrequencyType() error {
+	if opts.exportFrequencyType != "" {
+		if opts.exportFrequencyType != "daily" && opts.exportFrequencyType != "weekly" && opts.exportFrequencyType != "monthly" {
+			return errors.New("incorrect value for parameter exportFrequencyType. Value must be daily, weekly, or monthly")
+		}
+	}
+	return nil
+}
+
+func (opts *UpdateOpts) verifyReferenceHourOfDay(cmd *cobra.Command) error {
+	if cmd.Flags().Changed(flag.ReferenceHourOfDay) {
+		if opts.referenceHourOfDay < 0 || opts.referenceHourOfDay > 23 {
+			return errors.New("incorrect value for parameter referenceHourOfDay. Value must be an integer between 0 and 23 inclusive")
+		}
+	}
+	return nil
+}
+
+func (opts *UpdateOpts) verifyReferenceMinuteOfHour(cmd *cobra.Command) error {
+	if cmd.Flags().Changed(flag.ReferenceMinuteOfHour) {
+		if opts.referenceMinuteOfHour < 0 || opts.referenceMinuteOfHour > 59 {
+			return errors.New("incorrect value for parameter referenceMinuteOfHour. Value must be an integer between 0 and 59 inclusive")
+		}
+	}
+	return nil
+}
+
+func (opts *UpdateOpts) verifyRestoreWindowDays(cmd *cobra.Command) error {
+	if cmd.Flags().Changed(flag.RestoreWindowDays) {
+		if opts.restoreWindowDays <= 0 {
+			return errors.New("incorrect value for parameter restoreWindowDays. Value must be a positive, non-zero integer")
+		}
+	}
+	return nil
+}
+
+func checkForExport(out *atlas.CloudProviderSnapshotBackupPolicy) {
+	if out.Export == nil {
+		out.Export = new(atlas.Export)
+	}
+}
+
+func returnValueForSetting(enableFlag, disableFlag bool) *bool {
+	var valueToSet bool
+	if enableFlag && disableFlag {
+		return nil
+	}
+	if enableFlag {
+		valueToSet = true
+		return &valueToSet
+	}
+	if disableFlag {
+		valueToSet = false
+		return &valueToSet
+	}
+	return nil
+}
+
 func UpdateBuilder() *cobra.Command {
 	opts := &UpdateOpts{}
 	cmd := &cobra.Command{
 		Use:     "update",
 		Aliases: []string{"updates"},
 		Short:   "Update a snapshot backup policies for a cluster.",
+		Example: fmt.Sprintf(`  The following updates a snapshot backup policies for a cluster Cluster0:
+  $ %s backup schedule update --clusterName Cluster0 --updateSnapshots --exportBucketId 62c569f85b7a381c093cc539 --exportFrequencyType monthly`, cli.ExampleAtlasEntryPoint()),
 		PreRunE: func(cmd *cobra.Command, args []string) error {
 			preRun := opts.PreRunE(
 				opts.ValidateProjectID,
 				opts.initStore(cmd.Context()),
 				opts.InitOutput(cmd.OutOrStdout(), updateTemplate),
 			)
+			err := opts.verifyExportFrequencyType()
```

Comment:
How about using `opts.PreRunE(..., opts.verifyExportFrequencyType())`? It enforces all validations currently.

Reply:
@fmenezes Makes more sense that my implementation. I will update the code.